.github/workflows/docs-edit-automation.yml

```
@@ -0,0 +1,180 @@
+name: Documentation edit automation
+
+on:
+  push:
+    branches:
+      - 'docs-enhancement/**'
+      - 'docs-bugfix/**'
+
+permissions:
+  contents: write
+  pull-requests: write
+
+jobs:
+  auto-update-docs:
+    # Skip if pushed by bot to avoid loops
+    if: github.actor != 'github-actions[bot]'
+    runs-on: ubuntu-latest
+    
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
```

`v5` has been released recently, token is not needed when usin gthe default GH token.

```suggestion
        uses: actions/checkout@v5
```